### ISSUE TITLE
/v1/search endpoint handle's TLDs inconsistently.

### ISSUE THREAD

**Issue by AC-FTW** [blockstack/blockstack-core#847]
If I search the /v1/search endpoint for a fully qualified ID featuring the '.id' TLD, the search will fail--for instance try these searches:
- https://core.blockstack.org/v1/search?query=alexc.id
- https://core.blockstack.org/v1/search?query=judecn.id
- https://core.blockstack.org/v1/search?query=aaron.id

If I strip the '.id' TLD from those searches, the endpoint returns results:
- https://core.blockstack.org/v1/search?query=alexc
- https://core.blockstack.org/v1/search?query=judecn
- https://core.blockstack.org/v1/search?query=aaron

However when I search for a fully qualified ID featuring any other TLD that doesn't end in '.id' (e.g. '.id.blockstack'), the endpoint does return results:
- https://core.blockstack.org/v1/search?query=batzdorff.id.blockstack
- https://core.blockstack.org/v1/search?query=pie_head.id.blockstack

**Comment by jcnelson** [comment]
cc @kantai

**Comment by kantai** [comment]
Okay -- will update.

The behavior will be:

```
if (isNotFullyQualified(name)) {
   try `${name}.id`
} else {
   try name
}
```

**Comment by jcnelson** [comment]
Thanks @kantai!

**Comment by AC-FTW** [comment]
Thank you both @kantai & @jcnelson. 👍

**Comment by kantai** [comment]
This behavior should be better now -- try these examples:

https://core.blockstack.org/v1/search?query=alexc.id
https://core.blockstack.org/v1/search?query=judecn.id
https://core.blockstack.org/v1/search?query=aaron.id

**Comment by kantai** [comment]
Merged to master in #850 -- and live now. Those above queries should work. The additional complexity came because we wanted to keep the `username` field unchanged from prior behavior, otherwise this change would have been backward-incompatible.

**Comment by AC-FTW** [comment]
Great fix @kantai.  I've tested it with the names you provided above along with some others (prabhaav.stealthy, prabhaav.stealthy.id, pie_head, pie_head.id, pie_head.id.block, pie_head.id.blockstack) and it's working. Thanks 👍
